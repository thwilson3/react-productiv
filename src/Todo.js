import React from "react";

/** Simple presentation component for a todo.
 *
 * Props:
 * - todo: like { id, title, description, priority }
 *
 * { EditableTodo, TopTodo } -> Todo
 **/

function Todo({todo}) {

  const {title, priority, description} = todo;
  return (
      <div id={`todo-${todo.id}`} className="Todo">
        <div><b>{title}</b> <small>(priority: {priority})</small></div>
        <div><small>{description}</small></div>
      </div>
  );
}

export default Todo;
