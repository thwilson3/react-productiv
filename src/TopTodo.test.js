import React from "react";
import { fireEvent, render } from "@testing-library/react";
import TodoApp from "./TodoApp"
import Todo from "./Todo"
import TopTodo from "./TopTodo";
import { initialTodos } from "./App";

describe("TopTodo", function () {
    it("renders without crashing", function () {
        // this is a low-value test, but better than nothing
        render(<TopTodo todos={initialTodos} />);
    });

    it("shows the highest priority todo", function () {
        render(<TopTodo todos={initialTodos} />);
        const { container } = render(<Todo todo={initialTodos[1]} />);
        expect(container).toBeInTheDocument()
    });

    it("doesn't render if no todos", function () {
        const { container } = render(<TopTodo todos={[]} />)
        expect(container).toBeEmptyDOMElement();
    });

    it("matches snapshot", function () {
        const { container } = render(<TopTodo todos={initialTodos} />);
        expect(container).toMatchSnapshot();
    });
});