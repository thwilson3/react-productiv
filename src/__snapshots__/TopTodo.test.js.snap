// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo matches snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="todo-2"
  >
    <div>
      <b>
        Make dinner
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        Cook something healthy
      </small>
    </div>
  </div>
</div>
`;
